In [ ]:
import pandas as pd
from sklearn.metrics import mean_absolute_error, mean_squared_error, r2_score
from sklearn.linear_model import LinearRegression
# features

'''
Forecasted Load   'forecasted_load'
actual load       'actual_load'
Day Ahead Energy  Price 'day_ahead_lmp'
Day of Week       seperate out day of week from 'Date'
Hour of Day       seperate out hour of day from 'Date'
Month             seperate out month(have 12 months) from 'Date'
other_states_LMP  'AveragePrice_OtherStates'


'''
# from that find the actual energy price
# do with actual energy price 'actual_lmp'
# use regression model
df = pd.read_csv('merged_data2.csv')
df.drop_duplicates(inplace=True)

# Data preparation so that it works
df['Date'] = pd.to_datetime(df['Date'])
df['day_of_week'] = df['Date'].dt.dayofweek  #Monday=0,...,Sunday=6
df['hour_of_day'] = df['Date'].dt.hour
df['month'] = df['Date'].dt.month

features = [
    'forecasted_load',
    'actual_load',
    'day_ahead_lmp',
    'AveragePrice_OtherStates',
    'day_of_week',
    'hour_of_day',
    'month'
]

target = 'actual_lmp'



X = df[features]
y = df[target]

# Creating model
model = LinearRegression()
model.fit(X, y)

y_pred = model.predict(X)

df['predicted_lmp'] = y_pred

# Getting errors to analyze
mae = mean_absolute_error(y, y_pred)
rmse = mean_squared_error(y, y_pred)
r2 = r2_score(y, y_pred)


print(f"MAE={mae}")
print(f"RMSE= {rmse}")
print(f"R2= {r2}")


MAE=5.736087222991016
RMSE= 174.67584718668334
R2= 0.8126994903998349


For the above we did a simple linear model with the features stated above, from that we tried to predict the actual energy prices. From that we then got a linear prediction and then calculated some metrics like MAE, RMSE, and R^2 by comparing with the actual data. Our MAE was 5.736 which is the mean absolute error, which for us 5.73 is really good, if we look at the RMSE its 174.675 which is really high meaning that there is a crazy amount of variance, this means that maybe the linear model is not a good predictorfor our actual data and works well for some of it. if we look at the r^2 we can see that its .813 which is also good.